### PR TITLE
Support Logstash 6.x metrics

### PR DESCRIPTION
**Pull request by Evesy**
## Pull Request Checklist

#25 
#### General

- [x] Update Changelog following the conventions laid out [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [x] Update README with any necessary configuration snippets

- [ ] RuboCop passes

- [x] Existing tests pass

#### Purpose

See linked issue for details.

A continuation of https://github.com/sensu-plugins/sensu-plugins-logstash/pull/27

### CODE REVIEW

File: bin/metrics-logstash-node.rb [hunk 1]
```diff
@@ -113,31 +113,63 @@ def run # rubocop:disable Metrics/AbcSize
...
+      node['pipeline']['plugins']['outputs'].each do |item|
+        item['events'] = {} unless item.key?('events')
+        metrics["pipeline.plugins.outputs.#{item['name']}.#{item['id']}.events.in"] = item['events']['in'].to_i || 0
+        metrics["pipeline.plugins.outputs.#{item['name']}.#{item['id']}.events.out"] = item['events']['out'].to_i || 0
+        metrics["pipeline.plugins.outputs.#{item['name']}.#{item['id']}.events.duration_in_millis"] = item['events']['duration_in_millis'].to_i || 0
+      end
+    # logstash >= 6.0
+    elsif node.key?('pipelines')
+      node['pipelines'].each_key do |pipeline|
+        unless node['pipelines'][pipeline]['events'].nil?
```
Comment (Evesy) [inline bin/metrics-logstash-node.rb]: Can't assume that `node['pipelines'][pipeline]['events']` will always be a valid object, I've observed this to occasionally be a string value of `null`

Is this the correct way to handle this?

File: bin/metrics-logstash-node.rb [hunk 2]
```diff
@@ -113,31 +113,63 @@ def run # rubocop:disable Metrics/AbcSize
...
+          metrics["pipelines.#{pipeline}.plugins.filters.#{item['name']}.#{item['id']}.events.out"] = item['events']['out'].to_i || 0
+          metrics["pipelines.#{pipeline}.plugins.filters.#{item['name']}.#{item['id']}.events.duration_in_millis"] = item['events']['duration_in_millis'].to_i || 0 # rubocop:disable Metrics/LineLength
+          metrics["pipelines.#{pipeline}.plugins.filters.#{item['name']}.#{item['id']}.matches"] = item['matches'].to_i if item.key?('matches')
+        end
+
+        node['pipelines'][pipeline]['plugins']['outputs'].each do |item|
+          item['events'] = {} unless item.key?('events')
+          metrics["pipelines.#{pipeline}.plugins.outputs.#{item['name']}.#{item['id']}.events.in"] = item['events']['in'].to_i || 0
+          metrics["pipelines.#{pipeline}.plugins.outputs.#{item['name']}.#{item['id']}.events.out"] = item['events']['out'].to_i || 0
+          metrics["pipelines.#{pipeline}.plugins.outputs.#{item['name']}.#{item['id']}.events.duration_in_millis"] = item['events']['duration_in_millis'].to_i || 0 # rubocop:disable Metrics/LineLength
```
Comment (Evesy) [inline bin/metrics-logstash-node.rb:172]: Added this as I noticed it was missing from #27

**Comment by Evesy** [comment]
@runningman84 @majormoses I've carried on the work started in https://github.com/sensu-plugins/sensu-plugins-logstash/pull/27/files here

File: bin/metrics-logstash-node.rb (hunk 1 above)
Comment (majormoses) [reply bin/metrics-logstash-node.rb]: if if it falls into these three categories:
- string (valid)
- string 'null' (invalid)
- nil (invalid)

I think you need do some extra stuff...

```ruby
if node['pipelines'][pipeline]['events'].nil? || node['pipelines'][pipeline]['events'] == 'null'
  # skip me
else
  # do foo
end

``` 

That being said there are far too many nested conditionals here so the code is kinda unwieldy and will be slow, I think we need to break some of this out to additional functions.

**Comment by majormoses** [comment]
@Evesy thanks for that I will review it shortly.

**Review by majormoses — APPROVED** [review]
LGTM

Thanks for walking this over the line!

I am gonna check out the fork and ensure that all authors involved get credit [following this](https://help.github.com/articles/creating-a-commit-with-multiple-authors)

**Comment by majormoses** [comment]
After CI passes I will merge and release this.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-logstash/versions/2.1.0

**Pull request merged**